app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ *
+ */
+
+package com.wire.android.ui.home.conversations.details.editguestaccess
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.navigation.EXTRA_CONVERSATION_ID
+import com.wire.android.navigation.EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED
+import com.wire.android.navigation.EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED
+import com.wire.android.navigation.EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.QualifiedIdMapper
+import com.wire.kalium.logic.feature.conversation.UpdateConversationAccessRoleUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@HiltViewModel
+class EditGuestAccessViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+    private val dispatcher: DispatcherProvider,
+    private val updateConversationAccessRole: UpdateConversationAccessRoleUseCase,
+    savedStateHandle: SavedStateHandle,
+    qualifiedIdMapper: QualifiedIdMapper,
+) : ViewModel() {
+
+    val conversationId: QualifiedID = qualifiedIdMapper.fromStringToQualifiedID(
+        checkNotNull(savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)) {
+            "No conversationId was provided via savedStateHandle to EditGuestAccessViewModel"
+        }
+    )
+    private val isGuestAccessAllowed: Boolean = savedStateHandle.get<String>(EXTRA_EDIT_GUEST_ACCESS_IS_GUEST_ACCESS_ALLOWED)!!.toBoolean()
+    private val isServicesAllowed: Boolean = savedStateHandle.get<String>(EXTRA_EDIT_GUEST_ACCESS_IS_SERVICES_ALLOWED)!!.toBoolean()
+    private val isUpdatingGuestAccessAllowed: Boolean =
+        savedStateHandle.get<String>(EXTRA_EDIT_GUEST_ACCESS_IS_UPDATING_GUEST_ACCESS_ALLOWED)!!.toBoolean()
```

Comment:
yes we can, but why fetching the same value twice from database ? we already get it from previous screen

Reply:
Because it's less complicated and less error-prone, as you can see that you already had troubles with passing and retrieving navigation params 😄 Also, it wouldn't require to pass all these params when opening this screen - it can retrieve these values itself and we don't have to worry about it 😉 
We could extract the whole `observeConversationDetails` to a separate use case and make use of `.shareIn()` with replay so it works like a cache and don't have to query the DB every time, when passing `SharingStarted.WhileSubscribed()` we can set a timeout so it would still replay the same value for the next screen's ViewModel

Reply:
Okay..But the main point of passing data via navigation params is to avoid flickering, even with your approach the screen will flicker because screen state are initialised with default values then updated.

Also I don't see any raison to observe something not observable, `isGuestAccessAllowed` and `isServicesAllowed` does not change in real time if there is a change in a different client.

Reply:
If we use it the right way with the replay parameter set, then it'll get the proper value as the initial one already, so there should be no flickering. 😎 I believe we already had this discussion on collective and opted out of passing more data as navigation params because flickering was fixed by using right dispatchers. 😉 
And you're right, these value cannot be changed in the meantime by something else, but we still however for some reason observe them in `GroupConversationDetailsViewModel`, we could also pass data to this screen as navigation params, so why don't we do that? 😄 
It makes the code and already complex navigation more complicated in my opinion and makes this new screen dependent on the parent screen that opens it. Only crucial data (like IDs) that's used to retrieve all other data needed on a particular screen should be passed as a navigation params, you can read about it here: https://developer.android.com/jetpack/compose/navigation#retrieving-complex-data